<robot name="bravo7">
    
  <material name="blue">
    <color rgba="0.3 0.3 0.3 1"/>
  </material>

 <link name="mobile_base">
    <visual>
            <geometry>
                <mesh filename="package://bravo_description/meshes/caracara_visual_color.dae" scale="0.5 0.5 0.5"/>
            </geometry>
            <material name="blue"/>
        </visual>
  </link>

    <link name="base_link"/>
    <material name="bravo_black">
    <color rgba="0.3 0.3 0.3 0.7"/>
    </material>
    <joint name="bravo_joint" type="fixed">
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="base_link"/>
        <child link="bravo_base_link"/>
    </joint>
    <link name="bravo_base_link"/>

 <link name="world"/>   

    <joint name="loose_joint" type="fixed">
    <origin xyz="1 1 1" rpy="0 0 0"/>
    <parent link="world"/>
    <child link="mobile_base"/>
</joint>

 <joint name="bravo_base" type="fixed">
    <parent link="mobile_base"/>
    <child link="base_link"/>
    <origin rpy="3.14 0 0" xyz="-0.28 -0.5 -0.2"/>
  </joint>

    <link name="link1">
        <visual>
            <geometry>
                <mesh filename="package://bravo_description/meshes/RS2-266.stl"/>
            </geometry>
            <material name="bravo_black"/>
        </visual>
    </link>

    <joint name="base_fixed" type="fixed">
        <parent link="bravo_base_link"/>
        <child link="link1"/>
        <origin rpy="0 0 0"/>
    </joint>

    <link name="link2">
        <visual>
            <geometry>
                <mesh filename="package://bravo_description/meshes/RS2-217.stl"/>
            </geometry>
            <material name="bravo_black"/>
        </visual>
    </link>

    <joint name="joint1" type="revolute">
        <parent link="link1"/>
        <child link="link2"/>
        <origin rpy="0 0 0" xyz="0.0665 0 0.078"/>
        <axis xyz="0 0 -1"/>
        <limit effort="9.0" lower="0.0" upper="6.3" velocity="0.5"/>
    </joint>

    <link name="link3">
        <visual>
            <geometry>
                <mesh filename="package://bravo_description/meshes/RS2-250-212-214.stl"/>
            </geometry>
            <material name="bravo_black"/>
        </visual>
    </link>

    <joint name="joint2" type="revolute">
        <parent link="link2"/>
        <child link="link3"/>
        <origin rpy="0 0 0" xyz="-0.046 0 0.0674"/>
        <axis xyz="0 1 0"/>
        <limit effort="9.0" lower="0.0" upper="3.5" velocity="0.5"/>
    </joint>

    <link name="link4">
        <visual>
            <geometry>
                <mesh filename="package://bravo_description/meshes/RS2-268.stl"/>
            </geometry>
            <material name="bravo_black"/>
        </visual>
    </link>

    <joint name="joint3" type="revolute">
        <parent link="link3"/>
        <child link="link4"/>
        <origin rpy="0 0 0" xyz="-0.0052 0 -0.29055"/>
        <axis xyz="0 1 0"/>
        <limit effort="9.0" lower="0.0" upper="3.5" velocity="0.5"/>
    </joint>


    <link name="link5">
        <visual>
            <geometry>
                <mesh filename="package://bravo_description/meshes/RS2-214.stl"/>
            </geometry>
            <material name="bravo_black"/>
        </visual>
    </link>

    <joint name="joint4" type="revolute">
        <parent link="link4"/>
        <child link="link5"/>
        <origin rpy="0 0 0" xyz="0.0408 0 0.09695"/>
        <axis xyz="0 0 -1"/>
        <limit effort="9.0" lower="0.0" upper="6.3" velocity="0.5"/>
    </joint>

    <link name="link6">
        <visual>
            <geometry>
                <mesh filename="package://bravo_description/meshes/RS2-161.stl"/>
            </geometry>
            <material name="bravo_black"/>
        </visual>
    </link>

    <joint name="joint5" type="revolute">
        <parent link="link5"/>
        <child link="link6"/>
        <origin rpy="0 0 0" xyz="-0.0408 0 0.063"/>
        <axis xyz="0 1 0"/>
        <limit effort="9.0" lower="0.0" upper="3.5" velocity="0.5"/>
    </joint>



    <link name="link7">
        <visual>
            <geometry>
                <mesh filename="package://bravo_description/meshes/RS2-180.stl"/>
            </geometry>
            <material name="bravo_black"/>
        </visual>
    </link>

    <link name="force_torque_sensor">
        <visual>
            <geometry>
                <cylinder length="0.06" radius="0.04"/>
            </geometry>
            <material name="bravo_black"/>
        </visual>
    </link>

    <joint name="FT_fixed_1" type="fixed">
    <origin rpy="0 0 1.57" xyz=" 0 0 -0.03"/>
    <parent link="link7"/>
    <child link="force_torque_sensor"/>
    </joint>

    <joint name="joint6" type="revolute">
        <parent link="link6"/>
        <child link="link7"/>
        <origin rpy="0 3.14159 0" xyz="-0.0408 0 -0.14863"/>
        <axis xyz="0 0 -1"/>
        <limit effort="9.0" lower="0.0" upper="6.3" velocity="0.5"/>
    </joint>

    <joint name="bravo_jaws_joint" type="fixed">
    <origin rpy="0 0 0" xyz=" 0 0 0.124"/>
    <parent link="link7"/>
    <child link="bravo_jaws_base_link"/>
  </joint>

    <material name="bravo_jaws_black">
    <color rgba="0.3 0.3 0.3 1"/>
  </material>

  <link name="bravo_jaws_base_link"/>       
   
    

<!-- contact_point_joint -->
  <joint name="contact_point_joint" type="fixed">
<!-- 0.16/0.185 for plastic/metal tool with boardmarker-->
<!-- 0.15/0.21 for blue/white polymer tool with sponge -->
    <origin rpy="0 0 0" xyz="0 0 0.315"/>
    <parent link="force_torque_sensor"/>
    <child link="contact_point"/>
  </joint>

<!-- contact_point -->
  <link name="contact_point">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
      <material name="Red"/>
    </visual>
  </link>


</robot>
